src/BloomExe/Publish/PublishView.cs

```diff
@@ -466,7 +484,21 @@ private void OnPrint_Click(object sender, EventArgs e)
 			Logger.WriteEvent("Calling Print on PDF Viewer");
 			Analytics.Track("Print PDF");
 		}
+		private void OnPrintProgress(object sender, PdfPrintProgressEventArgs e)
+		{
+			// BL-788 Only called in Linux version.  Protects against button
+			// pushes while print is in progress.
+			if (e.PrintInProgress)
+			{
+				SetDisplayMode(PublishModel.DisplayModes.Printing);
+			}
```

Comment:
Do you need the UpdateDisplay() in both branches of the if, or only in the else branch as it is now?